### ISSUE TITLE
HTML constructor should only accept AbstractString

### ISSUE THREAD

**Issue by clarkevans** [JuliaLang/julia#38540]
The `Base.HTML` datatype seems an ideal wrapper to indicate that a given string object is suitable for direct presentation to a browser. However, currently the `HTML` type's [constructor](https://github.com/JuliaLang/julia/blob/5b7699217267cec91d5b2b3d1ce1d47fd53f6068/base/docs/utils.jl#L26)  accepts any sort of object for its argument and does not use the object's `text/html` representation.

```
function HTML(xs...)
    HTML() do io
        for x in xs
            print(io, x)
        end
    end
end
```

By printing, without providing the `text/html` mimetype, the `HTML` constructor is not using the hypertext representation of an object, as one would expect.  Indeed, it doesn't even handle nesting of itself in a reasonable way.  

```
julia> fragment = html"<span>Hello!</span>"
HTML{String}("<span>Hello!</span>")

julia> display("text/html", fragment)
<span>Hello!</span>

julia> display("text/html", HTML(fragment))
HTML{String}("<span>Hello!</span>")
```

This inability to handle itself (or other objects with hypertext rendering) poses a challenge when trying to use `HTML` as a unit of composition between various libraries and components that produce HTML. 

_A conservative option is to limit the `HTML` constructor to only `AbstractString`._ 

Long term, a coherent design rationale for this datatype and supporting operations could be worked out and implemented. For example, `join` could be made to accept an array of `HTML` objects and providing an `HTML` output of the concatenated `content` strings.

As more and more applications of Julia involve deep interactions with a web browser, having a standardized unit of composition for hypertext fragments would be immensely useful.  This ticket could be discussed in a [discourse](https://discourse.julialang.org/t/rationale-operations-for-built-in-html-type/50585) thread.

**Comment by vtjnash** [comment]
Possibly related to PR #13256. Possible IO integration PoC implemented in PR #27430.

**Comment by clarkevans** [comment]
I clearly didn't understand how the `HTML` type even works.
